dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -1412,8 +1412,18 @@ private void killAllTasks() {
             }
         }
 
-        this.taskRetryCheckList.clear();
-        this.depStateCheckList.clear();
+        for (Integer taskId : this.taskRetryCheckList.keySet()) {
+            if (taskInstanceHashMap.containsRow(taskId)) {
+                this.taskRetryCheckList.remove(taskId);
+            }
```

Comment:
![image](https://user-images.githubusercontent.com/42576980/176850220-ce99c1fc-2328-4c2a-a23d-da7f74105079.png)
I think it might be better to modify it like this

Reply:
updated